apps/client/src/features/cuesheet/CuesheetWrapper.tsx

```diff
@@ -69,37 +74,90 @@ export default function CuesheetWrapper() {
         return;
       }
 
-      const sheetData = makeTable(headerData, rundown, userFields);
-      const csvContent = makeCSV(sheetData);
+      let fileName = '';
+      let url = '';
+
+      if (exportType === 'json') {
+        const jsonContent = JSON.stringify({
+          headerData,
+          rundown,
+          userFields,
+        });
+
+        fileName = 'ontime export.json';
 
-      const blob = new Blob([csvContent], { type: 'text/csv;charset=utf-8;' });
-      const url = URL.createObjectURL(blob);
+        const blob = new Blob([jsonContent], { type: 'application/json;charset=utf-8;' });
+        url = URL.createObjectURL(blob);
+      } else if (exportType === 'csv') {
+        const sheetData = makeTable(headerData, rundown, userFields);
+        const csvContent = makeCSV(sheetData);
+
+        fileName = 'ontime export.csv';
+
+        const blob = new Blob([csvContent], { type: 'text/csv;charset=utf-8;' });
+        url = URL.createObjectURL(blob);
+      } else {
+        console.error('Invalid export type: ', exportType);
+        return;
+      }
 
       const link = document.createElement('a');
       link.setAttribute('href', url);
-      link.setAttribute('download', 'ontime export.csv');
+      link.setAttribute('download', fileName);
       document.body.appendChild(link);
       link.click();
-
       // Clean up the URL.createObjectURL to release resources
       URL.revokeObjectURL(url);
+      return;
     },
     [rundown, userFields],
   );
 
+  const onModalClose = (exportType?: ExportType) => {
+    setIsModalOpen(false);
+
+    if (!exportType) {
+      return;
+    }
+
+    if (headerData) {
+      exportHandler(headerData, exportType);
+    }
+  };
+
+  const handleOpenModal = (projectData: ProjectData) => {
+    setheaderData(projectData);
+    setIsModalOpen(true);
+  };
+
   if (!rundown || !userFields) {
     return <Empty text='Loading...' />;
   }
 
   return (
     <div className={styles.tableWrapper} data-testid='cuesheet'>
-      <CuesheetTableHeader handleCSVExport={exportHandler} featureData={featureData} />
+      <CuesheetTableHeader handleExport={handleOpenModal} featureData={featureData} />
       <Cuesheet data={rundown} columns={columns} handleUpdate={handleUpdate} selectedId={featureData.selectedEventId} />
+      <Modal isOpen={isModalOpen} onClose={onModalClose} motionPreset='scale' size='xl' colorScheme='blackAlpha'>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader fontWeight={400}>Choose your format</ModalHeader>
+          <ModalCloseButton />
+          <ModalBody className={styles.modalBody}>
+            <Button onClick={() => onModalClose('csv')} variant='ontime-filled' width='48%'>
+              CSV
+            </Button>
+            <Button onClick={() => onModalClose('json')} variant='ontime-filled' width='48%'>
+              JSON
+            </Button>
+          </ModalBody>
+        </ModalContent>
+      </Modal>
```

Comment:
Few nitpicks:

We would likely prefer isolating this in its own component.
Since we also implemented this both here and rundown it would also help with maintenance

The premise of the modal "Choose your format" is incorrect in the sense that the datasets are different, something more generic like "Download options" is maybe a better fit.
In the modal the user should be given two download options "Download project file" "Download CSV export" (terrible wording from my side too)